### PR TITLE
feat: make the testnode port allocation dynamic

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Dynamic port allocation for  the testnode as per https://github.com/celestiaorg/celestia-app/pull/1342#issuecomment-1425944402

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by evan-forbes — COMMENTED** [review]
can we also update these?
https://github.com/SweeXordious/celestia-app/blob/f20c5fdec8c7249434fc5f80bc99b55ed3236282/testutil/testnode/full_node_test.go#L43-L48

**Comment by SweeXordious** [comment]
sure, thanks for noting them

File: testutil/testnode/full_node.go [hunk 1]
```diff
@@ -209,3 +218,15 @@ func DefaultNetwork(t *testing.T, blockTime time.Duration) (cleanup func() error
 		return cleanupGRPC()
 	}, accounts, cctx
 }
+
+func getFreePort() int {
```
Comment (SweeXordious) [inline testutil/testnode/full_node.go:222]: [note for reviewers]
This is taken from here:

https://github.com/celestiaorg/celestia-node/blob/ef59637c0cf3f9f01ff587b37fa842123e641dc7/core/testing.go#L192-L202

**Review by evan-forbes — APPROVED** [review]
noice

**Comment by SweeXordious** [comment]
Any idea why this one is hanging?
https://github.com/celestiaorg/celestia-app/actions/runs/4145528714/jobs/7170074662
Is this happening lately?

**Review by MSevey — APPROVED** [review]

**Pull request merged**